packages/documentsearch/src/providers/notebooksearchprovider.ts

```diff
@@ -180,6 +180,36 @@ export class NotebookSearchProvider implements ISearchProvider {
     return this._currentMatch;
   }
 
+  async replaceCurrentMatch(newText: string): Promise<boolean> {
+    const notebook = this._searchTarget.content;
+    const editor = notebook.activeCell.editor as CodeMirrorEditor;
+    let replaceOccurred = false;
+    if (this._currentMatchIsSelected(editor)) {
+      const cellIndex = notebook.widgets.indexOf(notebook.activeCell);
+      const { provider } = this._cmSearchProviders[cellIndex];
+      replaceOccurred = await provider.replaceCurrentMatch(newText);
+      if (replaceOccurred) {
+        this._currentMatch = provider.currentMatch;
+        if (this._currentMatch) {
+          return Promise.resolve(replaceOccurred);
+        }
+      }
+    }
+    await this.highlightNext();
+    return Promise.resolve(replaceOccurred);
+  }
+
+  async replaceAllMatches(newText: string): Promise<boolean> {
+    let replaceOccurred = false;
+    for (let index in this._cmSearchProviders) {
+      const { provider } = this._cmSearchProviders[index];
+      const singleReplaceOccurred = await provider.replaceAllMatches(newText);
+      replaceOccurred = singleReplaceOccurred ? true : replaceOccurred;
+    }
+    this._currentMatch = null;
+    return Promise.resolve(replaceOccurred);
```

Comment:
Again, just return the value, and it will be wrapped in a promise since the function is declared async.